src/Datagrid/Pager.php

```diff
@@ -940,7 +940,23 @@ protected function initializeIterator()
             ), E_USER_DEPRECATED);
         }
 
-        $this->results = $this->getResults();
+        if (method_exists($this, 'getCurrentPageResults')) {
+            $results = $this->getCurrentPageResults();
+        } else {
+            @trigger_error(sprintf(
+                'Not implementing "%s::getCurrentPageResults()" is deprecated since sonata-project/admin-bundle 3.x and will fail in 4.0.',
+                PagerInterface::class
+            ), E_USER_DEPRECATED);
+
+            $results = $this->getResults();
+        }
+
+        if (\is_array($results)) {
```

Comment:
can `$results` also be a Doctrine Collection?

Reply:
It was previously an array, now it's an `iterable`.

A Doctrine Collection works with iterator_to_array.
And `$this->results` should be an array for BC. (But it's deprecated anyway)